src/electron/lib/api/apis/UtilApi.ts

```diff
@@ -43,7 +44,10 @@ export class UtilApi implements ElectronMainApi<UtilApi> {
       if (setting.secret) {
         setSecret(setting.id, setting.value.toString());
       } else {
-        settings.set(setting.value.toString(), setting.value);
+        if (setting.type === "preferences") {
```

Comment:
This implementation was a bit rushed when I pushed to get it finished in time for Demo 3. Looks it was a bit of an implementation error on my side which caused confustion. The basic idea here is to check whether the setting being saved is a secret (needs encryption) or whether it can just be stored as plain text. So, according to the only thing the else part should cotain is:
`settings.set(setting.id, setting.value);`
I don't seed the need for aditional if checks for various types of data.